### PR TITLE
raftstore/coprocessor: use scan way when approximate way fail

### PR DESCRIPTION
**Pull request by Connor1996**
## What have you changed? (mandatory)
cherry pick #3942 

## What are the type of the changes? (mandatory)
- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: src/raftstore/store/worker/split_check.rs [hunk 1]
```diff
@@ -227,15 +227,37 @@ impl<C: Sender<Msg>> Runner<C> {
                 let res = host.approximate_split_keys(region, &self.engine);
                 if let Err(e) = res {
                     error!(
-                        "[region {}] failed to get approxiamte split key: {}",
+                        "[region {}] failed to get approximate split key: {}, try scan way",
                         region_id, e
                     );
-                    return;
+                    let timer = CHECK_SPILT_HISTOGRAM.start_coarse_timer();
```
Comment (siddontang) [inline src/raftstore/store/worker/split_check.rs]: duplicated code with the above scan policy.

Comment (Connor1996) [reply src/raftstore/store/worker/split_check.rs]: yeah, it is just a quick fix to build a patch. I will change it.

**Comment by siddontang** [comment]
any test?

Btw, we must figure out why the approximate size is 0.

**Comment by Connor1996** [comment]
The failed test is not related to this PR. Merged.

**Pull request merged**